kernel/kernel-api/src/main/java/io/delta/kernel/data/FilteredColumnarBatch.java

```diff
@@ -108,4 +125,22 @@ public Row next() {
       public void close() {}
     };
   }
+
+  /**
+   * Returns the pre-computed number of selected rows if available.
+   *
+   * <p>If a selection vector is not present, it implies all rows are selected, and the method
```

Comment:
I think some simpler method docs could be something along the lines of:

```java
<p>If present, the number of selected rows was pre-computed and is retuned with no additional computation cost. This will happen in two cases: First, when the selection vector is empty and so we know the number of selected rows is equal to the batch size; and Second when the number of selected rows is .....
```

Basically, focus on what we are returning, instead of focusing on the selection vector